src/components/search/SearchResult/index.js

```diff
@@ -11,8 +11,11 @@ module.exports = React.createClass({
   },
 
   clickHandler: function(event) {
-    event.preventDefault();
-    this.props.onSelect(this.props.result);
+    var selectAction = this.props.onSelect(this.props.result);
+
+    if (selectAction) {
```

Comment:
I'd refactor the selectAction handling from `PageSearchModal`...At the moment I think selectAction will always be falsey? `undefined` because of no specified return value or `null` when the else is evaluated from `PageSearchModal`'s handler.